### DISCUSSION TITLE
Issues facing with CRDs upgrade to v1beta2

### DISCUSSION THREAD

**Discussion by naresh399** [strimzi/strimzi-kafka-operator#6187] [Q&A]
Hi Team,

I am on strimzi version 0.22.0 and I am trying to upgrade CRDs to v1beta2 and i wanted to upgrade version to 0.27.0

but when i am Changing CRDs to v1beta2. I am facing this issue

Error: UPGRADE FAILED: [unable to recognize "": no matches for kind "Kafka" in version "kafka.strimzi.io/v1beta2", unable to recognize "": no matches for kind "KafkaTopic" in version "kafka.strimzi.io/v1beta2", unable to recognize "": no matches for kind "KafkaUser" in version "kafka.strimzi.io/v1beta2"] 

Can you guys please check this one....

**Comment by naresh399** [comment]
I Ran this command and its showing only these. Why its not showing v1beta2?I am on version 0.22 now.
kubectl get crd kafkas.kafka.strimzi.io -o jsonpath="{.spec.versions[*].name}"
v1beta1 v1alpha1

**Reply by scholzj** [reply]
Have you done the API conversion using the provided tool as described in all the docs and release notes? If not, please have a look at it and give it a try.

**Reply by naresh399** [reply]
Hi scholzj, but we are using a helm artifactory image of 0.22 version.

**Reply by naresh399** [reply]
I have manually changed the API version in the yaml files. I didn’t use any conversion tools.

**Reply by scholzj** [reply]
> Hi scholzj, but we are using a helm artifactory image of 0.22 version.

Sorry, I don't follow. You talked about upgrading from 0.22 to 0.27. So I'm not sure what do you mean now with this.

> I have manually changed the API version in the yaml files. I didn’t use any conversion tools.

You have to follow the docs to do the conversion.

**Reply by naresh399** [reply]
Yes. I am already on version 0.22 and before upgrading to 0.27 I wanted to change API versions in Yaml files, so I did it manually with out using any tool because I am using Helm. So when I changed API version to v1beta2 in CRDs and deployed I saw that error.

**Reply by naresh399** [reply]
I follow the doc to do the conversion

there are 2 ways one is with tool and other one is manual way.

I choose the manual way.

**Reply by scholzj** [reply]
Well, I'm not sure what exactly you did. But it is quite hard to convert the resources manually. You can quite easily convert the custom resources manually. But as said in the documentation - you have to always follow it by the upgrade to the CRDs stored versions which cannot be done easily manually without using some tool (such as the API Conversion tool).

**Reply by naresh399** [reply]
![D124BBAC-2B23-416C-A5DA-B196E0E8A8EF](https://user-images.githubusercontent.com/33295675/149624638-b9feeaaa-3a7a-440d-bea3-e076078b32f1.jpeg)

This is what I was exactly planning to do. Upgrading to 0.22 then change the custom resources to v1beta2 and then plan to upgrade to 0.22

but I stopped where converting custom resources to v1beta2 with that error above I mentioned.

Can I use API conversion tool even if I am using helm?

**Reply by scholzj** [reply]
Helm has nothing to do with the CRD upgrade. Helm doesn't even update the CRDs when no conversion is needed. So you have to always do it manually in anyway. 

But in any case you have to use the conversion tool regardless how you installed Strimzi - whether from YAMLs, OperatorHub or Helm.
